File: merge-str-alternate/sol.go
func mergeAlternately(word1 string, word2 string) string {
    var b strings.Builder
    b.Grow(len(word1) + len(word2))
    i, cnt := 0, 0
    var char byte
    for i < len(word1) && i < len(word2) {
        if cnt % 2 == 0 {
            char = word1[i]
        } else {
            char = word2[i]
            i++
        }
        b.WriteByte(char)
        cnt++
    }

    if len(word1) > len(word2) {
        b.WriteString(word1[i:])
    } else if len(word2) > len(word1) {
        b.WriteString(word2[i:])
    }

    return b.String()
}

File: merge-str-alternate/sol.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        res = []

        for a, b in zip(word1, word2):
            res.append(a)
            res.append(b)
        
        res = "".join(res)
        if (len(word1) - len(word2)) != 0:
            if len(word1) > len(word2):
                res += word1[len(word2):]
            else:
                res += word2[len(word1):]

        return res

